astropy/table/tests/test_table.py

```diff
@@ -1894,14 +1894,15 @@ def test_simple_2(self):
         st = self.SimpleTable()
         dtypes = [np.int32, np.float32, np.float16]
         names = ['a', 'b', 'c']
-        t = table.Table(st, dtype=dtypes, names=names, meta=OrderedDict([('c', 3)]))
+        meta = OrderedDict([('c', 3)])
+        t = table.Table(st, dtype=dtypes, names=names, meta=meta)
         assert t.colnames == names
         assert all(col.dtype.type is dtype
                    for col, dtype in zip(t.columns.values(), dtypes))
 
-        # The supplied meta is ignored.  This is consistent with current
-        # behavior when initializing from an existing astropy Table.
-        assert t.meta == st.meta
+        # The supplied meta is overrides the existing meta.  Changed in astropy 3.2.
```

Comment:
This is the only required change in existing tests, which is fixing what seems like a bug.  (Supplied meta kwarg in Table init, where data is a Table, was always ignored).

Reply:
I agree this was a bug, but let's be sure to add it to `CHANGES.rst` in both the bug and api sections.